include/ts/ts.h

```diff
@@ -683,6 +686,9 @@ const char *TSUrlHttpParamsGet(TSMBuffer bufp, TSMLoc offset, int *length);
     copies the string to within bufp, so you can modify or delete
     value after calling TSUrlHttpParamsSet().
 
+    This function is deprecated. The value passed will be internally appended to the path portion.
+    Thus, TSUrlHttpParamsGet will not return the value set by this function.
```

Comment:
I would like to see us remove the TSUrlHttpParamsGet() instead of passing back an empty string.   People should change the logic in their plugins based on the change instead of it continuing to build and possibly discovering an issue later in production.

Reply:
Although that is a valid option, that would make the migration to 10.0 hard (plugins have to be changed to upgrade ATS). And we'd have to change the Lua API, Cripts, and url_sig plugin.

Everybody has to change the logic in their plugins at some point, if they use Get/SetParams, but this deprecation gives grace period.

What issues would be found? The APIs are not working as expected in the first place. And this change fixes a common mistake under the hood without fixing plugins.